homeassistant/components/synology_srm/sensor.py

```diff
@@ -0,0 +1,141 @@
+"""Sensor for Synology SRM routers.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/device_tracker.synology_srm/
+"""
+import synology_srm
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_MONITORED_CONDITIONS,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_SSL,
+    CONF_USERNAME,
+    CONF_VERIFY_SSL,
+)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+CONF_TRAFFIC_INTERVAL = "traffic_interval"
+
+DEFAULT_NAME = "synology_srm"
+DEFAULT_USERNAME = "admin"
+DEFAULT_PORT = 8001
+DEFAULT_SSL = True
+DEFAULT_VERIFY_SSL = False
+
+EXTERNALIP_CONDITION = "core.ddns_extip"
+TRAFFIC_CONDITION = "core.ngfw_traffic"
+
+POSSIBLE_MONITORED_CONDITIONS = {
+    "base.encryption",
+    "base.info",
+    EXTERNALIP_CONDITION,
+    "core.ddns_record",
+    "core.system_utilization",
+    "core.network_nsm_device",
+    TRAFFIC_CONDITION,
+    "mesh.network_wanstatus",
+    "mesh.network_wifidevice",
+    "mesh.system_info",
+}
+
+DEFAULT_MONITORED_CONDITIONS = [
+    EXTERNALIP_CONDITION,
+]
+
+POSSIBLE_TRAFFIC_INTERVAL = {"live", "day", "week", "month"}
+
+DEFAULT_TRAFFIC_INTERVAL = [
+    "live",
+]
+
+PLATFORM_SCHEMA = cv.PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_USERNAME, default=DEFAULT_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Optional(CONF_SSL, default=DEFAULT_SSL): cv.boolean,
+        vol.Optional(CONF_VERIFY_SSL, default=DEFAULT_VERIFY_SSL): cv.boolean,
+        vol.Optional(CONF_TRAFFIC_INTERVAL, default=DEFAULT_TRAFFIC_INTERVAL): vol.All(
+            cv.ensure_list, [vol.In(POSSIBLE_TRAFFIC_INTERVAL)]
+        ),
+        vol.Optional(
+            CONF_MONITORED_CONDITIONS, default=DEFAULT_MONITORED_CONDITIONS
+        ): vol.All(cv.ensure_list, [vol.In(POSSIBLE_MONITORED_CONDITIONS)]),
+    }
+)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Synology SRM Sensor."""
+    add_devices([SynologySrm(config)])
```

Comment:
This does not allow to automatically add a new `SynologySrm` sensor entity automatically when a new device connects to the router.

See #30334 `freebox/device_tracker` or #35412 synology_srm/device_tracker

Reply:
This is not a device tracker; this is to obtain information from the router; the existing SRM integration can be used for device tracking.

Reply:
I know this is not a device_tracker, but the behavior is the same.
You can actually see it in `icloud/sensor`.

But since when you have a new device connected to your router, you have to create a sensor, this should work like this.